x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingDeciderService.java

```diff
@@ -354,6 +357,23 @@ public AutoscalingDeciderResult scale(Settings configuration, AutoscalingDecider
                 .max()
                 .orElse(0L));
 
+        // This is an exceptionally weird state
+        // Our view of the memory is stale or we have tasks where the required job memory is 0, which should be impossible
+        if (largestJob == 0L && ((dataframeAnalyticsTasks.isEmpty() || anomalyDetectionTasks.isEmpty()) == false)) {
```

Comment:
The above code should make this scenario impossible. It makes no sense to have a `0` largest job and unempty tasks. I assume some bad state here and log a warning and return a no_scale.